gobblin-modules/gobblin-kafka-09/src/main/java/gobblin/source/extractor/extract/kafka/KafkaSimpleStreamingSource.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package gobblin.source.extractor.extract.kafka;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Properties;
+
+import org.apache.kafka.clients.consumer.Consumer;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.common.PartitionInfo;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.Throwables;
+import com.google.common.io.Closer;
+import com.google.common.base.Joiner;
+
+import gobblin.configuration.ConfigurationKeys;
+import gobblin.configuration.State;
+import gobblin.configuration.WorkUnitState;
+import gobblin.metrics.Tag;
+import gobblin.source.extractor.DataRecordException;
+import gobblin.source.extractor.RecordEnvelope;
+import gobblin.source.extractor.extract.EventBasedSource;
+import gobblin.source.workunit.Extract;
+import gobblin.source.workunit.WorkUnit;
+import gobblin.source.extractor.Extractor;
+import gobblin.configuration.SourceState;
+
+
+/**
+ * A {@link Source} implementation for a simple streaming kafka extractor.
+ *
+ * @author Shrikanth Shankar
+ *
+ */
+public class KafkaSimpleStreamingSource extends EventBasedSource<String, RecordEnvelope<byte[]>> {
+
+  private static final Logger LOG = LoggerFactory.getLogger(KafkaSimpleStreamingSource.class);
+
+  public static final String TOPIC_WHITELIST = "gobblin.streaming.kafka.topic";
+  public static final String TOPIC_NAME = "gobblin.streaming.topic.name";
+  public static final String PARTITION_ID = "gobblin.streaming.partition.id";
```

Comment:
As I like to say dots.are.not.substitutes.for.spaces. Please use dots only for scope separators. For example, gobblin.streaming.kafka.partitionId